### PR TITLE
Change activity crash

### PR DESCRIPTION
**Pull request by vincenzopalazzo**
This PR contains the code to fixes the issue #54.

The crash happened because during the init node state is not possible to change the activity because the change activity action stops the previous activity, such as MainActivity to run the LogActiviry or SettingActivity.

~~So, the solution is to make a more solid check of the node running and disable the setting and log menu item during the startup of the node.~~

In addition, I added propriety to MainActivity such as singleTop, this propriety force to use the instance of activity created, if exist. This resolved the UI bug when using the button in App bar to come back to MainActivity

In conclusion, I added also a little of clean up code!

### CODE REVIEW

**Comment by vincenzopalazzo** [comment]
The previous solution inside the commit https://github.com/lvaccaro/lamp/pull/55/commits/db89bedd18a3a13cc3d5484ade0b1c703d47bd4c was ugly for me and it doesn't work with all use cases.

Inside this commit https://github.com/lvaccaro/lamp/pull/55/commits/bd5999c890cdc8934f7f75d80a81f549aad770f8 I use the system notification to check if the node is running or is ready.

I-m happy to receive feedback when you have time :)

**Comment by vincenzopalazzo** [comment]
Rebased

**Comment by vincenzopalazzo** [comment]
This PR will be a review after join the PR https://github.com/lvaccaro/lamp/pull/62 to the master because in some use case this PR crash on Log View. But it is fixed already in https://github.com/lvaccaro/lamp/pull/62 :-)

**Comment by vincenzopalazzo** [comment]
According to our discussion, I'm removing the fix proposed because avoids the user using the UI, but the refactoring remains inside the propose.

To avoid confusion, I'm closing this PR because the object is no more fix the issue #54.
